tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -384,6 +384,7 @@ SpanBuffer SwapBuffers()
                 return null;
             }
 
+            trace = _statsAggregator?.ProcessTrace(trace) ?? trace;
```

Comment:
NIT - shall we start calling this a traceChunk to make it clearer everywhere that it may not be a complete trace?

```suggestion
            traceChunk = _statsAggregator?.ProcessTrace(traceChunk) ?? traceChunk;
```

Reply:
Good call. I have another PR where I'm doing some refactoring so I think I can make the change in that PR

Reply:
Actually, Lucas is beginning a refactor PR that is somewhat tangential. I think this would be a good time to do that